app/screens/image_preview/image_preview.js

```diff
@@ -370,10 +378,24 @@ export default class ImagePreview extends PureComponent {
                 return this.renderVideoPreview(file);
             }
 
+            const isGifImage = isGif(file.data);
+            if (file.data.has_preview_image) {
+                ImageCacheManager.cache(file.data.name, Client4.getFilePreviewUrl(file.data.id), this.updateFileImage);
+                return <Loading/>;
+            } else if (isGifImage) {
+                ImageCacheManager.cache(file.data.name, Client4.getFileUrl(file.data.id), this.updateFileImage);
+                return <Loading/>;
+            }
             return this.renderAttachmentIcon(file.data);
         }
 
         return <View/>;
+    });
+
+    updateFileImage = () => {
+        const {galleryKey} = this.state;
+        const keyIndex = parseInt(galleryKey.split('-')[1], 10);
+        this.setState({galleryKey: `gallery-${keyIndex + 1}`});
```

Comment:
I think you could just use a number for this key instead of parsing the string. The key doesn't need to be unique throughout the app, so the `gallery-` prefix is probably not necessary

Reply:
true, but just wanted to be sure it has meaning and parsing the string is not that big of a deal

Reply:
I guess that's fine although I don't think it's needed